modules/imgproc/test/test_filter.cpp

```diff
@@ -1923,25 +1923,25 @@ class CV_FilterSupportedFormatsTest : public cvtest::BaseTest
                 // non-separable filtering with a small kernel
                 fidx = 0;
                 cv::filter2D(src, dst, ddepth, small_kernel);
-                fidx++;
+                fidx = 1;
                 cv::filter2D(src, dst, ddepth, big_kernel);
-                fidx++;
+                fidx = 2;
                 cv::sepFilter2D(src, dst, ddepth, kernelX, kernelY);
-                fidx++;
+                fidx = 3;
                 cv::sepFilter2D(src, dst, ddepth, symkernelX, symkernelY);
-                fidx++;
+                fidx = 4;
                 cv::Sobel(src, dst, ddepth, 2, 0, 5);
-                fidx++;
+                fidx = 5;
                 cv::Scharr(src, dst, ddepth, 0, 1);
                 if( sdepth != ddepth )
                     continue;
-                fidx++;
+                fidx = 6;
                 cv::GaussianBlur(src, dst, Size(5, 5), 1.2, 1.2);
-                fidx++;
+                fidx = 7;
                 cv::blur(src, dst, Size(11, 11));
-                fidx++;
+                fidx = 8;
                 cv::morphologyEx(src, dst, MORPH_GRADIENT, elem_ellipse);
-                fidx++;
+                fidx = 9;
```

Comment:
What issue does it solve? I prefer the original version.

Reply:
Thank you for your review, this change comes from https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2019/p1152r4.html#expr.pre.incr .

Increment and decrement volatile variables, which are numeric and pointer, seems be deprecated for C++20.

Reply:
Perhaps this test should be rewritten using `ASSERT/EXPECT_NO_THROW` and `SCOPED_TRACE` :slightly_smiling_face:

Reply:
Thank you for your advice, I rewrite test with `EXPECT_NO_THROW` and `TEST_P` instead of loop with `SCOPED_TRACE`.